### DISCUSSION TITLE
postgres function failing with multiple arguments

### DISCUSSION THREAD

**Discussion by leo020120** [supabase/supabase#20551] [Questions] [answered]
For context, I have a React app and I'm trying to use rpc to run a function that does a simple insert into a table. 

The React function is below:

```
const confirmTeam = async (user, match, isHomeTeam) => {
    try {

      console.log("matchday", match);
      const { data, error: Error } = await supabase.rpc(
        "confirm_team_selection",
        {
          email: user.email,
          team: isHomeTeam ? match.homeTeamName : match.awayTeamName,
          game_week: match.gameWeek,
          match_id: match.matchId,
          userid: user.id,
          team_id: isHomeTeam ? match.homeTeamId : match.awayTeamId,
          team_location: isHomeTeam ? "HOME_TEAM" : "AWAY_TEAM",
          edition_id: activeEdition,
        }
      );

      if (Error) {
        console.error("Error updating existing active row:", Error);
        return;
      }

      console.log("Data", data);
    } catch (error) {
      console.log("error executing rpc", error);
    }

  };
```

And here is the postgres function i am creating:

```
create
or replace function confirm_team_selection (email text, team text, game_week bigint, match_id bigint, user_id uuid, team_id bigint, team_location text, edition_id bigint) returns table (result text) as $$

  -- start transaction
  BEGIN  

  --update existing active rows to inactive
  UPDATE mapping
  SET choice_status = 'inactive'
  WHERE user_id = userid
  AND choice_status = 'active';

  --insert the new row with active choice_status
  INSERT INTO mapping( email, team, game_week, match_id, user_id, team_id, team_location, edition_id, choice_status)
  VALUES(email, team, game_week, match_id, user_id, team_id, team_location, edition_id, 'active');

END;

$$ language plpgsql;
```

I have a total of 8 arguments that i want to include but when it runs i get a variation of the following error:

```
code: "PGRST202"
details: "Searched for the function public.confirm_team_selection with parameters edition_id, email, team_location, user_id or with a single unnamed json/jsonb parameter, but no matches were found in the schema cache."
hint: "Perhaps you meant to call the function public.confirm_team_selection(edition_id, match_id, team, userid)"
message: "Could not find the function public.confirm_team_selection(edition_id, email, team_location, user_id) in the schema cache"
```

In order to troubleshoot this i recreated the function to accept only 2 arguments (user_id and email), this worked ok so i then added another argument and this worked ok and i repeated this process until i got to around 5 arguments before it threw the error again. I then repeated the process but with the arguments in a different order to make sure that they were not causing the issue. By doing this i confirmed that they have all worked at some point and in some combination. For example, i know email works ok, but when i decide to add them in a different order, by time i add email as the 4th/5th/6th arg, that causes it to error, so i know its not the arguments or values themselves.

**Comment by GaryAustin1** [comment]
Sometimes when you change around functions types and order it can not be reflected in the schema cache the API uses.
Not sure you are encountering it or not as description of events is hard to follow.
Try running `NOTIFY pgrst, 'reload schema'` in the SQL editor.

**Reply by leo020120** [reply]
Thanks for your response, I tried reloading the schema cache as advised but i'm still getting the same behaviour.
I've tried to clarify the order of events below, hope it makes sense.

What i did was:

- Create the function (accepting all 8 arguments) and tested it. This failed with the error message i provided. 

- I then dropped the function and created it again, this time only using 'email' and 'user_id'. This worked ok.

- I thought perhaps one of the arguments in particular was the problem so i dropped this version of the function and created it again but this time including one more argument, so it now took email, user_id and team. This again was successful so i repeated the process of dropping and recreating the function, each time with one more argument to try and find one that made it fall over.

- Eventually it did fail, after adding 'team_id' i think. I couldn't see anything wrong with the team_id argument or the value it was pushing so i thought i'd test it. This time after dropping and recreating the function i Included team_id first with user_id and it worked.

- So i repeated the process of dropping and recreating the function, each time adding in a new argument but this time in a different order. However it then failed after adding an argument that definitely worked before, e.g. email.

So through this experimentation the function has inserted values from all of the arguments (in various combinations) but never all at once. The most it has worked with is 6 arguments but sometimes it fails after adding 4 or 5. It worth noting that when it fails the error message changes the arguments it says i'm trying to call with but this never reflects what i'm actually calling client side.

**Comment by steve-chavez** [comment, marked as answer]
```js
      console.log("matchday", match);
      const { data, error: Error } = await supabase.rpc(
        "confirm_team_selection",
        {
          email: user.email,
          team: isHomeTeam ? match.homeTeamName : match.awayTeamName,
          game_week: match.gameWeek,
          match_id: match.matchId,
          userid: user.id,
```

```sql
create
or replace function confirm_team_selection (email text, team text, game_week bigint, match_id bigint, user_id uuid
```

Your function cal doesn't have `userid` as a parameter it has `user_id`...
